fahrschein-opentracing/src/main/java/org/zalando/fahrschein/opentracing/OpenTracingHelper.java

```diff
@@ -0,0 +1,141 @@
+package org.zalando.fahrschein.opentracing;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import org.zalando.fahrschein.domain.Event;
+import org.zalando.fahrschein.domain.Metadata;
+
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import io.opentracing.propagation.TextMapExtractAdapter;
+import io.opentracing.propagation.TextMapInjectAdapter;
+import io.opentracing.tag.Tags;
+
+/**
+ * OpenTracing support class for Fahrschein Nakadi client. This class provides
+ * helper methods for setting up OpenTracing.
+ */
+public class OpenTracingHelper {
+
+	private final Tracer tracer;
+
+	private final String operation;
+
+	public OpenTracingHelper(Tracer tracer, String operation) {
+		this.tracer = tracer;
+		this.operation = operation;
+	}
+
+	/**
+	 * Converts the given nakadi specific representation to a span context that can
+	 * be used as parent for new spans.
+	 * 
+	 * @param tracer        the tracer
+	 * @param nakadiContext the nakadi context
+	 * @return The {@code SpanContext} for the given NakadiContext
+	 */
+	public static SpanContext convertNakadiContext(Tracer tracer, Map<String, String> nakadiContext) {
+		return tracer.extract(Format.Builtin.TEXT_MAP_EXTRACT, new TextMapExtractAdapter(nakadiContext));
+	}
+
+	/**
+	 * Converts the given span context into a Nakadi context that can be added to
+	 * the metadata of the Nakadi event.
+	 * 
+	 * @param tracer      the tracer instance
+	 * @param spanContext the span context
+	 * @return the nakadi context that can be added to the metadata of the Nakadi
+	 *         event
+	 */
+	public static Map<String, String> convertSpanContext(Tracer tracer, SpanContext spanContext) {
```

Comment:
actually I don't like this naming, because the question "convert to what?" is not obvious.

Reply:
`spanContextToMap`?

Reply:
renamed